drivers/temperature/adt7420/adt7420.c

```diff
@@ -97,25 +97,80 @@ int adt7420_reg_read(struct adt7420_dev *dev,
  * @return 0 in case of success, -1 otherwise.
 *******************************************************************************/
 int adt7420_reg_update_bits(struct adt7420_dev *dev,
-			    uint8_t register_address, uint8_t mask, uint8_t value)
+			    uint16_t register_address, uint8_t mask, uint8_t value)
 {
 
-	uint8_t data_buffer[3] = {0, 0, 0};
 	uint16_t regval;
 
 	if (adt7420_reg_read(dev, register_address, &regval))
 		return -1;
 
 	regval &= ~mask;
 	regval |= value;
-	if (adt7420_is_spi(dev) && no_os_field_get(ADT7320_L16, register_address)) {
-		data_buffer[0] = regval >> 8;
-		data_buffer[1] = regval & 0xFF;
+
+	return adt7420_reg_write(dev, register_address, regval);
+}
+
+/***************************************************************************//**
+ * @brief Write a value of a register via SPI
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
+ * @param data - Pointer to the register value
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int adt7420_spi_reg_write(struct adt7420_dev *dev, uint16_t register_address,
+			  uint32_t data)
+{
+	uint8_t data_buffer[3] = { 0, 0, 0 };
+	uint8_t num_bytes;
+
+	if (no_os_field_get(ADT7320_L16, register_address)) {
+		num_bytes = 3;
+		data_buffer[1] = (data >> 8);
+		data_buffer[2] = data & 0xFF;
+	} else {
+		num_bytes = 2;
+		data_buffer[1] = data;
+	}
+
+	/*Form the SPI register read command byte. Bits [5:3] of command byte
+	 holds the register address.*/
+	data_buffer[0] = (no_os_field_get(ADT7320_ADDR_MSK,
+					  register_address) << 3) & ADT7320_WRITE_MASK_CMD;
+	if (no_os_spi_write_and_read(dev->spi_desc, data_buffer, num_bytes))
+		return -1;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Write a value of a register via I2C
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
+ * @param data - Pointer to the register value
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int adt7420_i2c_reg_write(struct adt7420_dev *dev, uint16_t register_address,
+			  uint32_t data)
+{
+	uint8_t data_buffer[3] = { 0, 0, 0 };
+	uint8_t num_bytes;
+
+	if (no_os_field_get(ADT7320_L16, register_address)) {
+		num_bytes = 3;
+		data_buffer[2] = data & 0xFF;
+		data_buffer[1] = (data >> 8);
 	} else {
-		data_buffer[0] = regval;
+		num_bytes = 2;
+		data_buffer[1] = data;
 	}
 
-	return adt7420_reg_write(dev, register_address, data_buffer);
+	data_buffer[0] = register_address;
+
+	if (no_os_i2c_write(dev->i2c_desc, data_buffer, num_bytes, 1)) //no repeat start
```

Comment:
Same as prev comment, I think that you can return directly here.

Reply:
Returned directly